### PR TITLE
Add run command, refactor tests

### PR DESCRIPTION
**Pull request by mikeymike**
Hoping this is all right, I was behind master so rebased but I had to manually resolve some conflicts. I think it's okay but could be wrong :joy: 

Also got to double check Travis tests

### CODE REVIEW

**Comment by mikeymike** [comment]
darn it also dropped coverage :-1:

**Comment by mikeymike** [comment]
hmmm just Travis :trollface:

**Comment by mikeymike** [comment]
@AydinHassan I can't re-create the test failing locally on any PHP version. Can you pull down and run them. On Travis 7.0 failed then after re-run 7.0 passed and 5.6 failed. Pretty sure it was on the same test too.

File: src/Output/OutputInterface.php [hunk 1]
```diff
@@ -33,4 +35,14 @@ public function writeLine($line);
...
+
+    /**
+     * @return string
+     */
+    public function lineBreak();
+
+    /**
+     * @param Request $request
+     */
+    public function writeRequest(Request $request);
```
Comment (AydinHassan) [inline src/Output/OutputInterface.php]: This should type hint the PSR `RequestInterface`

File: src/Output/StdOutput.php [hunk 2]
```diff
@@ -38,6 +47,14 @@ public function printError($error)
     }
 
     /**
+     * @param $title
```
Comment (AydinHassan) [inline src/Output/StdOutput.php]: string

File: src/Output/StdOutput.php [hunk 3]
```diff
@@ -70,4 +87,50 @@ public function emptyLine()
...
+    /**
+     * @return string
+     */
+    public function lineBreak()
+    {
+        echo $this->color->__invoke(str_repeat("─", $this->terminal->getWidth()))->yellow();
+    }
+
+    /**
+     * @param Request $request
```
Comment (AydinHassan) [inline src/Output/StdOutput.php]: `RequestInterface`

File: src/Output/StdOutput.php [hunk 4]
```diff
@@ -70,4 +87,50 @@ public function emptyLine()
...
+     */
+    public function writeRequest(Request $request)
+    {
+        echo sprintf("URL:%s%s\n", str_repeat(' ', 5), $request->getUri());
```
Comment (AydinHassan) [inline src/Output/StdOutput.php]: remove the `str_repeat`. Just put it in the string.

File: src/ExerciseRunner/CgiRunner.php [hunk 5]
```diff
@@ -196,13 +196,19 @@ public function run($fileName, OutputInterface $output)
             $event      = $this->eventDispatcher->dispatch(new CgiExecuteEvent('cgi.run.usr-execute.pre', $request));
             $process    = $this->getProcess($fileName, $event->getRequest());
 
+            $output->writeTitle("Request\n");
```
Comment (AydinHassan) [inline src/ExerciseRunner/CgiRunner.php]: can we use `emptyLine()` ? `\n` looks a bit weird in there

**Comment by mikeymike** [comment]
@AydinHassan I'll rebase before merge if all is good :+1:

**Comment by mikeymike** [comment]
my bad didn't re-run the test :rabbit:

**Comment by AydinHassan** [comment]
@mikeymike This ready? Can we squash it?

**Comment by mikeymike** [comment]
Yeah can do, I did just realise I hadn't written a test for the RunCommand itself :expressionless: ... I can do this after the merge though, got nothing better to be doing :laughing: 

What do you think ?

**Comment by AydinHassan** [comment]
Yeah, fine by me :smile: That's probably why the coverage went down!

**Pull request merged**